### PR TITLE
Add ExecutionNode constructor xml comments

### PR DESCRIPTION
**Pull request by Shane32**
See #2064

### CODE REVIEW

**Comment by Shane32** [comment]
If the xml comments look good, I can copy them to the other constructors.

File: src/GraphQL/Execution/ExecutionNode.cs [hunk 1]
```diff
@@ -76,8 +76,19 @@ public object Source
...
+        /// for child nodes of an array execution node, not a <see cref="ListGraphType"/> instance. Typically
+        /// this should equal <see cref="FieldDefinition">fieldDefinition</see>.<see cref="FieldType.ResolvedType">ResolvedType</see>.<see cref="GraphQLExtensions.GetNamedType(IGraphType)">GetNamedType()</see></param>
+        /// <param name="field">The AST field of this node</param>
+        /// <param name="fieldDefinition">The graph's field type of this node</param>
+        /// <param name="indexInParentNode">For child array item nodes of a <see cref="ListGraphType"/>, the index of this array item within the field; otherwise, null</param>
         protected ExecutionNode(ExecutionNode parent, IGraphType graphType, Field field, FieldType fieldDefinition, int? indexInParentNode)
         {
+            if (graphType is NonNullGraphType)
+                throw new System.ArgumentOutOfRangeException(nameof(graphType), "For non-nullable graph types, graphType must represent the unwrapped graph type");
+            if (IndexInParentNode.HasValue && graphType is ListGraphType)
```
Comment (sungam3r) [inline src/GraphQL/Execution/ExecutionNode.cs:90]: What about list of lists ? 

[[Int]] : [[1], [1,2], [1,2,3]]

Comment (Shane32) [reply src/GraphQL/Execution/ExecutionNode.cs:90]: I forgot that was possible.  I suggest we remove the checks, but keep the comments (after review to be sure they are still correct).

Comment (sungam3r) [reply src/GraphQL/Execution/ExecutionNode.cs:90]: OK

File: src/GraphQL/Execution/ExecutionNode.cs [hunk 2]
```diff
@@ -76,8 +76,19 @@ public object Source
...
+                throw new System.ArgumentOutOfRangeException(nameof(graphType), "For children nodes of a list graph type, graphType must represent the unwrapped graph type");
```
Comment (Shane32) [inline src/GraphQL/Execution/ExecutionNode.cs:91]:
```suggestion
```

File: src/GraphQL/Execution/ExecutionNode.cs [hunk 3]
```diff
@@ -76,8 +76,19 @@ public object Source
         /// <summary>
         /// Initializes an instance of <see cref="ExecutionNode"/> with the specified values
         /// </summary>
+        /// <param name="parent">The parent node, or null if this is the root node</param>
+        /// <param name="graphType">The underlying graph type of this node; not a <see cref="NonNullGraphType"/>, and
+        /// for child nodes of an array execution node, not a <see cref="ListGraphType"/> instance. Typically
```
Comment (Shane32) [inline src/GraphQL/Execution/ExecutionNode.cs:81]:
```suggestion
        /// for child nodes of an array execution node, not a <see cref="ListGraphType"/> instance unless the children are themselves lists. Typically
```

File: src/GraphQL/Execution/ExecutionNode.cs [hunk 4]
```diff
@@ -76,8 +76,19 @@ public object Source
...
+        /// this should equal <see cref="FieldDefinition">fieldDefinition</see>.<see cref="FieldType.ResolvedType">ResolvedType</see>.<see cref="GraphQLExtensions.GetNamedType(IGraphType)">GetNamedType()</see></param>
```
Comment (Shane32) [inline src/GraphQL/Execution/ExecutionNode.cs:82]:
```suggestion
        /// <param name="graphType">The graph type of this node, unwrapped if it is a <see cref="NonNullGraphType"/>.
        /// Array nodes will be a <see cref="ListGraphType"/> instance.</param>
```

File: src/GraphQL/Execution/ExecutionNode.cs [hunk 5]
```diff
@@ -76,6 +76,12 @@ public object Source
         /// <summary>
         /// Initializes an instance of <see cref="ExecutionNode"/> with the specified values
         /// </summary>
+        /// <param name="parent">The parent node, or null if this is the root node</param>
+        /// <param name="graphType">The graph type of this node, unwrapped if it is a <see cref="NonNullGraphType"/>.
+        /// Array nodes will be a <see cref="ListGraphType"/> instance.</param>
```
Comment (Shane32) [inline src/GraphQL/Execution/ExecutionNode.cs:81]:
```suggestion
        /// <param name="graphType">The graph type of this node, unwrapped if it is a <see cref="NonNullGraphType"/>. Array nodes will be a <see cref="ListGraphType"/> instance.</param>
```

**Review by sungam3r — APPROVED** [review]

**Pull request merged**